AndroidAnnotations/androidannotations-rest-spring/rest-spring-test/src/main/java/org/androidannotations/rest/spring/test/MyService.java

```diff
@@ -141,94 +142,92 @@
 	@Post("http://company.com/client/ping")
 	void ping();
 
-	// There should be max 1 parameter that is not mapped to an attribute. This
-	// parameter will be used as the post entity.
 	@Post("/events/")
 	@Accept(MediaType.APPLICATION_JSON)
-	Event addEvent(Event event);
+	Event addEvent(@Body String event);
 
 	@Post("/events/{year}/")
-	Event addEvent(Event event, int year);
+	Event addEvent(@Body Event event, int year);
 
 	@Post("/events/{year}/")
 	Event addEvent(int year);
 
 	@Post("/events/")
-	ResponseEntity<Event> addEvent2(Event event);
+	ResponseEntity<Event> addEvent2(@Body Event event);
 
-	@Post("/events/{date}")
+	@Post("/events/{date}?myCookieInUrl={myCookieInUrl}")
```

Comment:
I was really surprised when I notice this, too. This is why I asked you [before](https://github.com/excilys/androidannotations/pull/1615#issuecomment-154435454). There was no validation that checks if `@RequiresCookieInUrl` parameter is in url or not. I had to fix these test cases because I added `hasAnnotatedAllParameters` method that validates the number of url variables in url is equals with the sum of non-annotated parameters, parameters with `@Path` and `@RequiresCookieInUrl`

...ok I felt I have to add more validations on `hasAnnotatedAllParameters`. 

Do you want to remove current `hasAnnotatedAllParameters` validation and send me another pull request for validating `@RequiresCookieInUrl`? which will block this pull request. It won't take that long.